casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -38,15 +38,21 @@ object Genesis {
                     wallets: Seq[Wallet],
                     startHash: StateHash,
                     runtimeManager: RuntimeManager)(implicit scheduler: Scheduler): BlockMessage = {
-
-    val blessedTerms = List(
+    val defaultBlessedTerms = List(
       LinkedList.term,
       NonNegativeNumber.term,
       MakeMint.term,
       BasicWallet.term,
       (new Rev(wallets)).term
     ).map(termDeploy)
-
+    withContracts(defaultBlessedTerms, initial, wallets, startHash, runtimeManager)
+  }
+  
+  def withContracts(blessedTerms: List[Deploy],
```

Comment:
It is a little against Scala style to give a default paremeter value by overloading the function explicitly. Maybe consider doing something like 
```scala

   def withContracts(initial: BlockMessage,
                     wallets: Seq[Wallet],
                     startHash: StateHash,
                     runtimeManager: RuntimeManager,
                     blessedTerms: List[Deploy] = List( ... ))(implicit scheduler: Scheduler): BlockMessage = { ... }
```

Reply:
You don't mind the long list and logic that will go in?

```
   def withContracts(initial: BlockMessage,
                     wallets: Seq[Wallet],
                     startHash: StateHash,
                     runtimeManager: RuntimeManager,
                     blessedTerms: List[Deploy] = List(
      LinkedList.term,
      NonNegativeNumber.term,
      MakeMint.term,
      BasicWallet.term,
      (new Rev(wallets)).term
    ).map(termDeploy))
```

Reply:
Actually nvm, we can't do it because of `wallets`.

Reply:
Oh yes, you're right, sorry. It's not just a default parameter because wallets are used in the other function. Probably fine as is then.